play-json/shared/src/main/scala/play/api/libs/json/JsMacroImpl.scala

```diff
@@ -566,6 +566,7 @@ import scala.reflect.macros.blackbox
     // --- Sub implementations
 
     val readsType = c.typeOf[Reads[_]]
+    val writesType = c.typeOf[Writes[_]]
```

Comment:
Refactor the sealed family macro,
- `writeLambda` consistent with `readLambda` (both using `inferImplicitValue`), whereas the `writeLambda` was using [`implicitly` in the generated tree before](https://github.com/playframework/play-json/pull/216/files#diff-e3cb35ba4b91adceabbb4a153abef895L624),
- no longer use (inefficient) ["shadowing workaround"](https://github.com/playframework/play-json/pull/216/files#diff-e3cb35ba4b91adceabbb4a153abef895L637).